main.tf

```diff
@@ -65,104 +84,31 @@ resource "google_compute_instance" "elasticsearch" {
 
   network_interface {
     network = var.network
-
-    access_config {
-      // Ephemeral IP
-    }
   }
 
   metadata = {
-    ssh-keys = "devops:${tls_private_key.provision.public_key_openssh}"
+    ssh-keys  = "devops:${tls_private_key.provision.public_key_openssh}"
+    user-data = <<-EOT
+#!/bin/bash
+
+echo "${base64encode(local.elasticsearch_configuration)}" | base64 -d > /tmp/elasticsearch.yml;
```

Comment:
Why do we need to do the Base64 encode + decode?

Reply:
I just needed a way how to send a file from local terraform to remote compute instance without fearing any bash magic (escape bash related characters, ...), I guess base64 was straight forward choice, I am open to suggestions